packages/nimble-components/src/table/specs/table-data-api.md

```diff
@@ -0,0 +1,142 @@
+# Table Data API
+
+## Problem Statement
+
+A table/data-grid component can have a variety of ways to introduce data into it. We should make it clear how we expect users to introduce data into the `nimble-table` and address the implications that has on the API.
+
+## Links To Relevant Work Items and Reference Material
+
+[FAST data-grid spec](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/data-grid/data-grid.spec.md) (see `rowsData` in _properties:_ section)
+
+[TanStack Table API](https://tanstack.com/table/v8/docs/api/core/table)
+
+## Implementation / Design
+
+In the `nimble-table`, the data to associated with rows will be exposed through a `data` property that is an array of key/value pairs. The key/value pairs within the data must have a key of type string and a value that is of a supported type. The supported value types are: `string`, `number`, `boolean`, `Date`, `null`, and `undefined`. As implied by the set of supported value types, complex value types such as arrays or objects are not supported. The supported value types are limited to ensure data operations are fast and to eliminate the need to custom sort functions, which could hurt performance. While the types are limited, a rendered column in the table can use data from multiple keys in a row's data item. The details about the way column definitions use data from the `data` property are out of scope for this spec.
+
+Because `data` is a complex type, it will not be exposed as an attribute on the `nimble-table` element. As a result, the data must be provided to the table programatically rather than declaratively with HTML.
+
+This API is similar to the `FASTDataGrid`. Its [data API](https://github.com/microsoft/fast/blob/416dc9167e9d41e6ffe11d87ed79b2f455357923/packages/web-components/fast-foundation/src/data-grid/data-grid.ts#L193), is provided via a `rowsData` property of type `object[]`.
+
+To help enforce typing, the `Table` class will be generic on the type for the row data. The typing of the table is shown below:
+
+```ts
+export class Table<
+    TData extends {
+        [key: string]: string | number | boolean | Date | null | undefined;
+    }
+> extends FoundationElement {
+    @observable
+    public data: TData[];
+}
+```
+
```

Comment:
Think it might be helpful to include a quick usage example. i.e. 

```ts
The user of a table can type a reference to the table as:
const tableRef: Table<TData>;
tableRef.data = [{}] // the data property will be bound by the above type
```